src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -353,12 +359,27 @@ private void uploadFleetStatusServiceData(Set<GreengrassService> greengrassServi
         }
         List<ComponentStatusDetails> components = new ArrayList<>();
         long sequenceNumber;
+
         synchronized (greengrassServiceSet) {
+
             // If there are no Greengrass services to be updated, do not send an update.
             if (greengrassServiceSet.isEmpty()) {
                 return;
             }
 
+            //When a component version is bumped up, FSS may have pointers to both old and new service instances
```

Comment:
component versions don't change the instance. The exact same instance is used

Reply:
True, but if a deployment adds component X with v1 and next deployment removes component X. Then if next  deployment adds a new version of X (v2). Then a new instance is created.